### DISCUSSION TITLE
Question - How to avoid `import bleak`  fail ?

### DISCUSSION THREAD

**Discussion by RomRom1** [hbldh/bleak#475] [Q&A] [answered]
# How to mock bleak properly ?  

* bleak version: 0.10
* Python version: 3.7
* Operating System: Linux
* BlueZ version (`bluetoothctl -v`) in case of Linux: bluetoothctl doesn't exist inside a docker container

### Description

I'm using bleak in my python library. 

I use `bleak.discover()` so at the beginning of my file I import bleak module
In my unit test, I mock `bleak.discover` to isolate them.

All works find on my laptop, but when CI runs all unit tests into a docker container (`python:3.7-slim-buster`) an exception is raised.

```
ImportError: Failed to import test module: test_pu_update_mode
Traceback (most recent call last):
  
  ...
  
  File "/builds/python-libraries/utils/********/lib/python3.7/site-packages/********.py", line 6, in <module>
    import bleak
  File "/builds/python-libraries/utils//********/lib/python3.7/site-packages/bleak/__init__.py", line 34, in <module>
    p = subprocess.Popen(["bluetoothctl", "--version"], stdout=subprocess.PIPE)
  File "/usr/local/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/local/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'bluetoothctl': 'bluetoothctl'
```

Of course `bluetoothctl` is not found because bluetoothctl is not available inside container.

### Workaround

As workaround, I provide in my CI an environment variable `READTHEDOCS` set to `True` to avoid this issue, but I'm not sure that is a good way.

### Question

What is the proper way to do not show import statement failed due to that ?
If it's a good way it is possible to add `READTHEDOCS` condition to Windows as well ?

**Comment by hbldh** [comment, marked as answer]
The `READTHEDOCS` flag skips the version check on the `bluez` package. Either use that (which is used for building documentation on Read the Docs) or install bluez in your container. [Bleak's CI workflows does precisely that](https://github.com/hbldh/bleak/blob/develop/.github/workflows/build_and_test.yml#L23-L25).

**Reply by RomRom1** [reply]
Sorry for delay. Bluez installing seems to be better. 

Thanks for your response.

**Reply by antonelliCP** [reply]
This is confusing to an Android newbie, like myself:
1) Is there a working version of BLEak for android?  
2) is the environment variable READTHEDOCS =1 a working patch? 
3) Is the bluez package needing to be updated on android tablets/phones to make this work?
